### PR TITLE
[documentation] #1280: Document Iroha metrics

### LINKED ISSUE #1280: Document Iroha metrics
At the moment the only way to get list of metrics is to request them from the prometheus port and read text response containing help comments like that:

```
> curl http://127.0.0.1:5551/metrics
# HELP blocks_height Total number of blocks in chain
# TYPE blocks_height gauge
blocks_height 135543
# HELP peers_number Total number peers to send transactions and request proposals
# TYPE peers_number gauge
peers_number 7
# HELP number_of_domains Total number of domains in WSV
# TYPE number_of_domains gauge
number_of_domains 14
# HELP total_number_of_transactions Total number of transactions in blockchain
# TYPE total_number_of_transactions gauge
total_number_of_transactions 216499
# HELP number_of_signatures_in_last_block Number of signatures in last block
# TYPE number_of_signatures_in_last_block gauge
number_of_signatures_in_last_block 5
```

It would be convenient to add and maintaing list of all metrics in iroha readthedocs' documentation

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

### Description of the Change

Add documentation for metrics: https://github.com/outoftardis/iroha/blob/doc-metrics/docs/source/guides/metrics.md

### Issue

Partially addresses #1280

### CODE REVIEW

File: docs/source/guides/metrics.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Metrics
+
+To conveniently and thoroughly monitor the performance of the network, you can use metrics: numeric data collected over time about your Iroha network.
```
Comment (appetrosyan) [inline docs/source/guides/metrics.md:3]:
```suggestion
To conveniently and thoroughly monitor the performance of the network, we recommend using [prometheus](https://prometheus.io/). It's a program that can monitor your Iroha peer over a separate socket, providing you different kinds of performance metrics. 
```

File: docs/source/guides/metrics.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+You can then analyse the data to make your project even more efficient.
```
Comment (appetrosyan) [inline docs/source/guides/metrics.md:5]:
```suggestion
This data can help find performance bottlenecks, and optimise Iroha's configuration. 
```

File: docs/source/guides/metrics.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## How to use metrics
+
+To use metrics, you need to add it to your [Iroha configuration](../references/config.md).
```
Comment (appetrosyan) [inline docs/source/guides/metrics.md:9]:
```suggestion
To use metrics, you need to configure a metrics endpoint in the [Iroha configuration](../references/config.md). By default the endpoint is exposed at `127.0.0.1:8180/metrics`, if the port is not available, Iroha will still start and work normally, but metrics won't be accessible.
```

**Review by appetrosyan — CHANGES_REQUESTED** [review]

**Comment by appetrosyan** [comment]
I think we need a little more detail on what metrics are gathered exactly. Perhaps we should add a subcommand to `kagami` to generate a `metrics` reference, what do you think @Arjentix ?

**Comment by Arjentix** [comment]
Still there is no way to read about all metrcis without requesting an end-point. So this PR doesn't close the issue, but maybe it shouldn't...

@appetrosyan , I don't know how exactly prometheus metrics are working in our project, but if it's possible to generate this reference, it would be very good.

After looking at code I think it might be possible with procedural macro and global variable magic. We can try it

**Review by Arjentix — APPROVED** [review]
I think the doc is alright and we can merge it now and update later, then there will be a reference generated by our `kagami` tool

**Review by appetrosyan — APPROVED** [review]

**Review by Arjentix — APPROVED** [review]

**Pull request merged**